app/src/main/java/org/schabi/newpipe/fragments/list/BaseListFragment.java

```diff
@@ -350,7 +350,8 @@ protected void showStreamDialog(final StreamInfoItem item) {
             return;
         }
 
-        final ArrayList<StreamDialogEntry> entries = new ArrayList<>();
+        final ArrayList<StreamDialogEntry> entries =
+                new ArrayList<>(Arrays.asList(StreamDialogEntry.show_channel_details));
```

Comment:
No idea, why the implementqtion was used here. We should go with the interface.
```suggestion
        final List<StreamDialogEntry> entries =
                new ArrayList<>(Arrays.asList(StreamDialogEntry.show_channel_details));
```

Reply:
And maybe also use `Collections.singletonList` instead of `Arrays.asList`

Reply:
@Stypox is there any reason in particular to do that?

Reply:
Arrays.asList creates a new List of fixed size, Collections.singletonList creates a custom List of size 1 where everything is optimized for that.

Reply:
Maybe, even better, just replace it with:
```suggestion
final List<StreamDialogEntry> entries =
                new ArrayList<>();
entries.add(StreamDialogEntry.show_channel_details);
```

Reply:
Oh, also, please put the open channel option at the bottom of the menu, it's less important than others

Reply:
@Stypox done.